File: Wanaheim/ClientApp/src/app/models/item.ts
import { Data } from "@angular/router";

export interface KeyValuePair {
    id: number;
    name: string;
}

export interface Item {
    id: number;
    name: string;
    price: number;
    quantity: number;
    description: string;
    creationDate: Data;

    category: KeyValuePair;
    subcategory: KeyValuePair;
}

export interface ItemSave {
    id: number;
    name: string;
    price: number;
    quantity: number;
    description: string;

    categoryId: number;
    subcategoryId: number;
}

File: Wanaheim/Core/Domain/Item.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Threading.Tasks;

namespace Wanaheim.Core.Domain
{
    public class Item 
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public decimal Price { get; set; }
        public int Quantity { get; set; }
        public string Description { get; set; }
        public DateTime CreationDate { get; set; }

        public ICollection<Photo> Photos { get; set; }

        public int SubcategoryId { get; set; }
        public Subcategory Subcategory { get; set; }

        public Item()
        {
            Photos = new Collection<Photo>();
        }
    }
}
